test/e2e/util_gatewayapi_test.go

```diff
@@ -590,6 +623,103 @@ func assertGatewaySuccessful(t *testing.T, namespace, name string) (*gatewayapiv
 	return gw, nil
 }
 
+func waitForGatewayListenerCondition(t *testing.T, gatewayName types.NamespacedName, listenerName string, conditions ...metav1.Condition) error {
+	t.Helper()
+
+	gateway := &gatewayapiv1.Gateway{}
+	expected := gatewayListenerConditionMap(conditions...)
+	current := map[string]string{}
+
+	err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 1*time.Minute, false, func(context context.Context) (bool, error) {
+		if err := kclient.Get(context, gatewayName, gateway); err != nil {
+			t.Logf("failed to get gateway %s: %v", gatewayName.Name, err)
+			return false, nil
+		}
+
+		var listenerStatus *gatewayapiv1.ListenerStatus
+		for _, ls := range gateway.Status.Listeners {
+			if string(ls.Name) == listenerName {
+				listenerStatus = &ls
+				current = gatewayListenerConditionMap(listenerStatus.Conditions...)
+				return conditionsMatchExpected(expected, current), nil
+			}
+		}
+
+		if listenerStatus == nil {
+			t.Logf("listener %s not found in gateway %s", listenerName, gatewayName.Name)
+			return false, nil
+		}
+		return false, nil
+	})
+
+	if err != nil {
+		t.Errorf("Expected conditions: %v\n Current conditions: %v", expected, current)
+		if gateway != nil {
+			if gwStatusPretty, err := json.MarshalIndent(gateway.Status, "", "  "); err != nil {
+				t.Fatal(err)
+			} else {
+				t.Logf("Gateway %s/%s status: %s\n", gateway.Namespace, gateway.Name, string(gwStatusPretty))
+			}
+		}
+	}
```

Comment:
We don't need to print out the gateway status.
```suggestion
	if err != nil {
		t.Errorf("Expected conditions: %v\n Current conditions: %v", expected, current)
	}
```